refactor(oxlint): run oxlint before tsgolint

camc314 opened the pull request:
This PR changes the order in which oxlint and tsgolint are run.
Previously, we run tsgolint, THEN oxlint, however in order to support
disable directives, this is going to be reversed, this will allow the
disable directives data to be parsed + stored for both the oxlint run,
and then the tsgolint run, and finally, we can report unused disable
directives

overlookmotel commented:
TSGolint doesn't read / touch the Rust ASTs or Semantic, does it? (including when receiving errors back from TSGolint)

When running JS plugins, it mutates the AST to change offsets from UTF-8 to UTF-16, and empties `Semantic`. Before this PR, JS plugins ran last, so didn't matter. But now TSGolint is last.

I don't *think* the order change should matter, but just checking.

camc314 commented:
> Will this hashmap be stored outside of Semantic, and is it possible for it to avoid referencing any data in Semantic?

yes and yes. chat tomo to explain my thinking?

camc314 committed:
refactor(oxlint): run oxlint before tsgolint (#13519)

This PR changes the order in which oxlint and tsgolint are run.
Previously, we run tsgolint, THEN oxlint, however in order to support
disable directives, this is going to be reversed, this will allow the
disable directives data to be parsed + stored for both the oxlint run,
and then the tsgolint run, and finally, we can report unused disable
directives
